pkg/client/logs.go

```diff
@@ -64,11 +68,6 @@ func NewReader(bytestream chan []byte, errc chan error) *Reader {
 // Read manages the message overflow ensuring no bytes are missed.
 // If an error is set on the reader it will return the error immediately.
 func (r *Reader) Read(p []byte) (int, error) {
-	// Always return the error if it is set.
-	if r.err != nil {
-		return 0, r.err
-	}
```

Comment:
We really want to always drain the overflow buffer; otherwise we could have messages which were received but only partially printed.

This is especially of interest now that the message queue could be empty but we receive an error which may need to be broken into parts/printed _then_ returned.